File: Leetcode/704.Binary_Search/C++/Binary_Search.cpp
#include <bits/stdc++.h>
#include<vector>
using namespace std;

class Solution {
public:

    int search(vector<int>& nums, int target) {
        vector<int>::iterator it;
        bool not_found=true;
        int size = nums.size();
        int mid=size/2;
        for(int i=0;i<mid;i++){
        	if(nums[i]==target){
        		not_found=false;
        		cout << "find " << target << " index: " << i<<endl;
        		return i;       		
			} 
		}
		for(int i=mid;i<size;i++){
        	if(nums[i]==target){
        		not_found=false;
        		cout << "find " << target << " index: " << i<<endl;
        		return i; 
			} 			
		}
		if(not_found){
			cout << "not found!!" << endl; 
			return -1;
		}
    }
    
};

int main(){
	vector<int> nums;
	Solution sol;
	int n;
	while(n != -2){
		cin >> n;
		nums.push_back(n);
	}
	cin >> n;
	sol.search(nums,n);
	
}

File: Leetcode/704.Binary_Search/JavaScript/binary_search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function (nums, target) {
  let i = 0;
  let j = nums.length - 1;
  while (i <= j) {
    let mid = Math.floor((i + j) / 2);
    if (target > nums[mid]) {
      i = mid + 1;
    } else if (target < nums[mid]) {
      j = mid - 1;
    } else if (target === nums[mid]) {
      return mid;
    }
  }
  return -1;
};

let nums = [-1, 0, 3, 5, 9, 12];
let target = 9;
console.log(search(nums, target));
